designer/client/src/data/condition/getConditions.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  hasConditionGroup,
+  hasConditionName,
+  type ConditionWrapper,
+  type FormDefinition
+} from '@defra/forms-model'
+
+import { type FieldDef } from '~/src/data/component/fields'
+
+/**
+ * @param data - the form definition data
+ * @param fields - an array of current fields
+ */
+export function getConditions(data: FormDefinition, fields: FieldDef[]) {
+  const { conditions } = data
+
+  conditions.forEach((condition: ConditionWrapper) => {
```

Comment:
Sorry @davidjamesstone, can we fix this so `data` isn't modified?

Currently the call to `getConditions(data)` has side effects on the original `data` state

This has caused us bugs with React state getting out of sync, skipping renders or other oddities

```js
const conditions = getConditions(data)
console.log(data) // Original has been modified 🚨
```

E.g. See `updateLinksTo()` where the original `data` is returned when changes are unnecessary:

https://github.com/DEFRA/forms-designer/blob/ff8700c6e0a4165caccc4ed66e48a5717dc5f317/designer/client/src/data/page/updateLinksTo.ts#L8-L16

Reply:
Resolved with ab09bf2b51ff968f8a33d0f15a19cfdfb250bc34

Reply:
Sorry to come back to this one

I've had a look at [`updateConditions()` in `ab09bf2`](https://github.com/DEFRA/forms-designer/commit/ab09bf2b51ff968f8a33d0f15a19cfdfb250bc34#diff-1cc829d1a36951e1a446bc0eb972b5cb846e7988e4103c9f6236cd2781a43af6R14-R18) and we're cloning the definition every time it's called, even when no fixes are required. Are you sure we can't address this?

E.g. Here's the test for `updateStartPage()` where we use `toStrictEqual()` to confirm the object was left untouched when no start page changes were necessary:

https://github.com/DEFRA/forms-designer/blob/055dd0c8451eb3d522aa1e21d865d06818beb35c/model/src/form/utils/update-start-page.test.ts#L139-L145

Reply:
Resolved with 8570a50eb10ad66109040f811c36e5284c143d75. We have to walk the tree to see if anything would change, if it doesn't return the data as-is, otherwise clone and alter.